README.md

```diff
@@ -140,6 +142,40 @@ If you'd like to have `LSP-metals` format your file on document save then make s
 ...
 ```
 
+### Organize imports on save
+
+If you'd like to have `LSP-metals` organize your imports on document save then make sure to add this to your `LSP` settings.
+
+```
+"lsp_code_actions_on_save":{
+  "source.organizeImports": true
+},
+...
+```
+
+### Decoration protocol
+
+This plugin implements the [Decoration protocol](https://scalameta.org/metals/docs/editors/decoration-protocol.html) (only for ST4) which allows showing worksheet evaluations (instead of comments), as well as inferred types and other info as Sublime Phantom.
+
+To enable this feature enable the following `LSP-metals` settings:
+
+```
+"initializationOptions": {
+  ...
+  "decorationProvider": true,
+  "inlineDecorationProvider": true
```

Comment:
Honestly, I'd just have these enabled by default. If not, you force the user to manually have to enable this here and in the settings. Afaik, there shouldn't be any negative effect of just turning these to `true` by default, and then letting the users decide if they want them shown via the relevant settings.

Reply:
I wasn't aware of that. I though that `inlineDecorationProvider` was taking precedence over the user setting for the server. 
The reason it is disabled by default is that it is difficult to make off for ST3 and on for ST4 by default. 
Anyway I will change that

Reply:
@ckipp01 I just did a test where `decorationProvider` and `inlineDecorationProvider` where true but `show-inferred-type`, `show-implicit-arguments`, `show-implicit-conversions-and-classes` where all false.

It is all fine in code but. But worksheet evaluation was still using decoration which doesn't work for ST3. Is there any worksheet setting that be set to force metals into using comments instead of relying on the provider flags ?

Reply:
There really isn't a setting for that. What I do in `coc-metals` is has logic that will determine if the user is using NeoVim (supports virtual text) or Vim (doesn't), and sets the value accordingly. Could you do something similar here and detect if the user is using ST3 or ST4?

Reply:
Ok I will handle the setting directly in the code depending on the client.